pkg/sideinput/interface.go

```diff
@@ -0,0 +1,18 @@
+package sideinput
+
+import (
+	"context"
+)
+
+type RetrieverSideInput interface {
+	// RetrieveSideInput is the function to process each side-input request.
+	RetrieveSideInput(ctx context.Context) MessageSI
```

Comment:
`Retrieve()`?

Reply:
Shall we give an error option? What do we expect the implementation to do if there's for sure an error will happen (maybe temporarily, but retry won't work)? In this case, what will be the returned []byte? and do we update the exiting data?

Reply:
Have added the error option to the function. In the manager side if there is an error here, then right now we just log it
```
	f := func() {
		if err := sim.execute(ctx, sideInputClient, siStore); err != nil {
			log.Errorw("Failed to execute the call to fetch Side Inputs.", zap.Error(err))
		}
	}
```